### PR TITLE
Implement "showcase nav" component

### LINKED ISSUE #17096: Implement "Showcase nav" component
# Feature

Implement the "Showcase nav" component in the quo2 component library.

### Designs

[Link to Figma](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=19006%3A205435&mode=design&t=w2rLGtX3APUF5Kn0-1)

### Acceptance Criteria:

- Should follow the [best practices](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md)
- Should add [component test](https://github.com/status-im/status-mobile/blob/develop/doc/component-tests-overview.md)
- Should add an export from the `quo2.core` namespace
- Add a preview for this component under the `profile` category in the Quo2 preview screen

### PR DESCRIPTION
**Pull request by smohamedjavid**
fixes #17096

### Summary

This PR implements the `Showcase nav` component which is needed for emoji picker development.

### Design

[Link to Figma](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=19006%3A205435&mode=design&t=w2rLGtX3APUF5Kn0-1)

### Platforms

- Android
- iOS

### Steps to test

- Open Status
- Navigate to `Quo2 Preview > profile > showcase nav`

status: ready

### CODE REVIEW

File: src/quo2/components/profile/showcase_nav/component_spec.cljs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+(ns quo2.components.profile.showcase-nav.component-spec
+  (:require [quo2.components.profile.showcase-nav.view :as view]
+            [reagent.core :as reagent]
+            [test-helpers.component :as h]))
+
+(def generate-previews
+  (->> (range 5)
+       (map inc)
+       (mapv (fn [index]
+               {:icon (rand-nth [:i/recent :i/profile :i/communities :i/wallet :i/nft :i/token
```
Comment (J-Son89) [inline src/quo2/components/profile/showcase_nav/component_spec.cljs]: do we want random data in our tests? 🤔

File: src/quo2/components/profile/showcase_nav/style.cljs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+(ns quo2.components.profile.showcase-nav.style
+  (:require [react-native.safe-area :as safe-area]
+            [quo2.foundations.colors :as colors]))
+
+(defn container
+  [state theme]
+  {:padding-bottom   (+ (safe-area/get-bottom) 12)
```
Comment (J-Son89) [inline src/quo2/components/profile/showcase_nav/style.cljs]: imo we don't want to add the padding/margin like this to the container of the component - rather a `container-style` prop and we set it from our uses across the app. `padding` perhaps is fine on containers however it seems odd to me to use safe-area within our design system components. 
wdyt? 🤔

File: src/quo2/components/profile/showcase_nav/component_spec.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/quo2/components/profile/showcase_nav/component_spec.cljs]: The component can display any Icon. So, I used random data.

File: src/quo2/components/profile/showcase_nav/style.cljs (hunk 2 above)
Comment (smohamedjavid) [reply src/quo2/components/profile/showcase_nav/style.cljs]: The component has the iOS home bar (which is the safe area) and that space (padding-bottom) is not needed for Android. So, I included it in the component. 

The actual padding bottom is 12px from the icon row to the safe area. hence, `(+ (safe-area/get-bottom) 12)`

Comment (J-Son89) [reply src/quo2/components/profile/showcase_nav/style.cljs]: sure, whatever you think is best 👍

File: src/quo2/components/profile/showcase_nav/component_spec.cljs (hunk 1 above)
Comment (J-Son89) [reply src/quo2/components/profile/showcase_nav/component_spec.cljs]: yeah sure, I understand it won't affect the test. However unless we explicitly need random data for these tests I think it is generally a better approach to have consistent data in our tests.

**Review by J-Son89 — APPROVED** [review]

File: src/quo2/components/profile/showcase_nav/component_spec.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/quo2/components/profile/showcase_nav/component_spec.cljs]: Right. I will update the tests.

File: src/quo2/components/profile/showcase_nav/style.cljs (hunk 2 above)
Comment (smohamedjavid) [reply src/quo2/components/profile/showcase_nav/style.cljs]: @J-Son89 - You are right. Components like the `bottom tab` don't rely on a safe area. The safe area is controlled by the parent component which uses it. I have updated the component to make it more independent. The `container-style` will allow us to update if required. 👍

File: src/quo2/components/profile/showcase_nav/component_spec.cljs (hunk 1 above)
Comment (smohamedjavid) [reply src/quo2/components/profile/showcase_nav/component_spec.cljs]: Updated the tests 👍

File: src/quo2/components/profile/showcase_nav/style.cljs (hunk 2 above)
Comment (J-Son89) [reply src/quo2/components/profile/showcase_nav/style.cljs]: ok great, thanks for the update!

**Review by Francesca-G — APPROVED** [review]
Here's the [review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3408%3A222910&mode=design&t=hxndcLu0NW4nMg3H-1) 
it looks good but please check the spacing, I left a comment there :)

**Comment by smohamedjavid** [comment]
Thank you for the review @Francesca-G.





Regarding the spacing, it's using 8px between the buttons (as described in Figma). I believe it's due to the icons, it give a different perception of the space between them.

**Pull request merged**